app/models/manager_refresh/inventory_collection/builder.rb

```diff
@@ -0,0 +1,273 @@
+module ManagerRefresh
+  class InventoryCollection
+    # @see test in /spec/models/manager_refresh/inventory_collection/builder_spec.rb
+    class Builder
+      class MissingModelClassError < StandardError; end
+
+      require_nested :CloudManager
+      require_nested :InfraManager
+      require_nested :AutomationManager
+      require_nested :NetworkManager
+      require_nested :StorageManager
+      require_nested :List
+
+      include ::ManagerRefresh::InventoryCollection::Builder::Shared
+
+      # Default options for builder
+      #   :adv_settings
+      #     - values from Advanced settings (doesn't overwrite values specified in code)
+      #     - @see method ManagerRefresh::Inventory::Persister.options()
+      #   :adv_settings_enabled
+      #     - enable/disable this properties
+      #   :auto_missing_parent
+      #     - auto assigns parent if this property is missing
+      #     - @see method add_parent_if_missing()
+      #   :auto_model_class
+      #     - tries to set model_class from persister class
+      #     - @see method auto_model_class
+      #   :auto_inventory_attributes
+      #     - auto creates inventory_object_attributes from target model_class setters
+      #     - attributes used in InventoryObject.add_attributes
+      #   :shared_properties
+      #     - any properties applied if missing (not explicitly specified)
+      #   :without_model_class
+      #     - if false and no model_class derived or specified, throws exception
+      def self.default_options
+        {
+          :adv_settings              => {},
```

Comment:
Does this contain things like is targeted refresh enabled?

Reply:
Basically - no. ```:shared_properties``` are intended for get targeted property
It contains information from settings - set by user:
https://github.com/slemrmartin/manageiq/blob/6971ccbeca86576f1fc866fe81381de1510195d9/app/models/manager_refresh/inventory/persister.rb#L79

Both shared_properties and adv_settings are applied to all inventory collections
- both are low-level properties (cannot rewrite concrete code's definitions)

Difference between shared_properties and adv_options:
- adv_options 
- - is filled by user in UI: Configuration -> Tab Adv. settings
- shared_options
- - is defined in persister, usually contains :strategy and :targeted properties per persister
- - rewrites adv_options properties

My answer is above. I doubt about meaning of adv_options - but I made refactoring 1:1 only

Reply:
Right and https://github.com/slemrmartin/manageiq/blob/6971ccbeca86576f1fc866fe81381de1510195d9/app/models/manager_refresh/inventory/persister.rb#L79 has things like allow_targeted_refresh: https://github.com/ManageIQ/manageiq-providers-amazon/blob/master/config/settings.yml#L88-L89

Just trying to figure out what settings are going where.

Reply:
Yes, these settings are not given to adv_settings, just subkey :inventory_collections

Reply:
> just subkey :inventory_collections

Ahhh okay this is what I was missing, that wasn't in the link you sent here https://github.com/slemrmartin/manageiq/blob/6971ccbeca86576f1fc866fe81381de1510195d9/app/models/manager_refresh/inventory/persister.rb#L79